### ISSUE TITLE
System.Management.Tests Time bug

### ISSUE THREAD

**Issue by SemperPeritus** [dotnet/corefx#26422]
With UTC+7 Tomsk (automatic daylight time was turned off) I get error. System.Management.Tests works fine with UTC-8 and turned on automatic daylight time. Test passes after changing time back to UTC+7 without *clean --all* through *build-tests*. After *clean --all*, *build* and *build-tests* error appears again.

I am using Windows 10 x64 and VS 2017.

Part of the log:
```
C:\Users\plato\source\repos\corefx\Tools\tests.targets(500,5): error : One or more tests failed while running tests fro m 'System.Management.Tests' please check C:\Users\plato\source\repos\corefx\bin/tests/System.Management.Tests/netcoreap p-Windows_NT-Debug-x64/testResults.xml for details! [C:\Users\plato\source\repos\corefx\src\System.Management\tests\Sys tem.Management.Tests.csproj]
```

*testResults.xml*:
```
      <test name="System.Management.Tests.ManagementDateTimeConverterTests.DateTime_RoundTrip" type="System.Management.Tests.ManagementDateTimeConverterTests" method="DateTime_RoundTrip" time="0.0306261" result="Fail">
        <failure exception-type="Xunit.Sdk.EqualException">
          <message><![CDATA[Assert.Equal() Failure\r\nExpected: 2002-04-08T14:18:35.9780000Z\r\nActual:   2002-04-08T15:18:35.9780000Z]]></message>
          <stack-trace><![CDATA[   at System.Management.Tests.ManagementDateTimeConverterTests.DateTime_RoundTrip() in C:\Users\plato\source\repos\corefx\src\System.Management\tests\System\Management\ManagementDateTimeConverterTests.cs:line 18]]></stack-trace>
        </failure>
      </test>
```

**Comment by SemperPeritus** [comment]
@danmosemsft, Is it the System.TimeZone bug? I am also have 7 hours offset by UTC, not 8.
![2018-01-19 4 _li 1](https://user-images.githubusercontent.com/11060914/35124290-255d717a-fcd8-11e7-8ccc-10c6ef9a98ec.jpg)

In **System.TimeZone.CurrentTimeZone.m_CachedDaylightChanges.Keys** Count = 0 at this point:
![2018-01-19](https://user-images.githubusercontent.com/11060914/35124524-f45e012e-fcd8-11e7-9fa2-7b60566ea56a.png)


And Count = 1 at this point:
![2018-01-19 2](https://user-images.githubusercontent.com/11060914/35124555-136b7a1a-fcd9-11e7-8322-cd80c7c0956f.png)

Is it OK?

**Comment by danmosemsft** [comment]
@krwq or @pjanotti would have to say

**Comment by pjanotti** [comment]
Fixed by PR #27652
